src/software/jetson_nano/thunderloop.cpp

```diff
@@ -168,7 +172,25 @@ Thunderloop::~Thunderloop() {}
                 world_ = new_world;
             }
 
+            auto nanoseconds_elapsed_since_last_world =
+                getNanoseconds(time_since_last_vision_received);
+
+            // Timeout Overrides for World and Primitives
+            // These should be after the new primitive / world update sections above
+
             // If world not received in a while, stop robot
+            if (nanoseconds_elapsed_since_last_world > PRIMITIVE_MANAGER_TIMEOUT_NS)
```

Comment:
nit: Can we change the timeout's name to be more general if we're using it for both Primitives and Worlds? e.g.
```suggestion
            if (nanoseconds_elapsed_since_last_world > PACKET_TIMEOUT_NS)
```